src/app/SharpRaven/Utilities/Serializer.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Text;
+
+namespace SharpRaven.Utilities
+{
+    public static class Serializer
+    {
+        public static byte[] SerialiseBytes(this object obj)
```

Comment:
The only reason I did that is I wasn't sure if there was extra information used in the connection side of thigns that was not sent as part of the json packet. Eg does the json serialised packet contain everything that is required to recreate it? Also I was not sure if you had a de-serialise method.

So basically I used an existing method I use in other projects as I was sure it would work and time was limited. If there exists a de-serialise method.

If jsonserialise serialises everything we need, and there is an easy to implement deserialise then yes it would be more appropriate.